src/include/duckdb/execution/index/art/prefix.hpp

```diff
@@ -13,15 +13,21 @@
 
 namespace duckdb {
 class Prefix {
+	static constexpr idx_t PREFIX_INLINE_BYTES = 8;
```

Comment:
I was wondering about this size. It seems too big. For integer types we only have a 8 byte prefix for `BIGINT`, and only if the ART consists of only one leaf as root. For `HUGEINT` this might occur for some prefixes, but I don't know if that is so relevant. Compound keys could have these long prefixes, and strings.

Maybe when indexing strings this makes more sense, but then the actual problem is more the sparsity of the alphabet with ASCII. This causes a lot of very small nodes (Node4) on lower levels. I think then you'd also want a much smaller inlined prefix. Otherwise these many nodes are taking up a lot of space, no?

So maybe a prefix inline size of 4 bytes or even only 2 bytes is better? Would be an interesting experiment... I have the data sets, but I have to see if I find the time to include it in my thesis. :)

Reply:
As explained in the meeting - the prefix is inlined where the pointer would normally be, and pointers are 8-bytes on 64-bit systems. We could reduce the size to 4 on 32-bit systems, but optimizing for 32-bit systems is probably not worth the complexity.